### PR TITLE
Added configuration of cert signing algorithms

### LINKED ISSUE #163: Safe Cert Gen SHA Feature
Request - add a feature to Safe to allow the operator on a cert gen to specify a SHA256 vs the default SHA512 when creating a cert.

### PR DESCRIPTION
**Pull request by thomasmmitchell**
Supports rsa with md5, sha1, sha256, sha384, and sha512.
Supports rsapss with sha256, sha384, and sha512
x509 show now shows the signing algorithm used. Supports
displaying even more algos than we actually support issuing,
because that was trivial. This configuration is available on
x509 issue, renew, and reissue.

Closes #163

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -2362,42 +2365,47 @@ Issue a new X.509 Certificate
...
+
+  -u, --key-usage     An x509 key usage or extended key usage. Can be specified
+                      once for each desired usage. Valid key usage values are:
+                      'digital_signature', 'non_repudiation', 'key_encipherment',
+                      'data_encipherment', 'key_agreement', 'key_cert_sign',
+                      'crl_sign', 'encipher_only', or 'decipher_only'. Valid
+                      extended key usages are 'client_auth', 'server_auth', 'code_signing',
+                      'email_protection', or 'timestamping'
+
+  -l, --sig-algorithm The algorithm that the certificate will be signed
```
Comment (dennisjbell) [inline main.go]: Please specify the default if not given

File: main.go [hunk 2]
```diff
@@ -2481,23 +2490,28 @@ Reissues an X.509 Certificate with a new key.
...
+                      originally signed the certificate, but does not
+                      have to be.
+
+  -t, --ttl           How long the new certificate will be valid
+                      for.  Specified in units h (hours), m (months)
+                      d (days) or y (years).  1m = 30d and 1y = 365d
+                      Defaults to the last TTL used to issue or renew
+                      the certificate.
+
+  -l, --sig-algorithm The algorithm that the certificate will be signed
```
Comment (dennisjbell) [inline main.go]: Please specify the default if not given

File: main.go [hunk 3]
```diff
@@ -2583,18 +2605,23 @@ Renew an X.509 Certificate with existing key
...
```
Comment (dennisjbell) [inline main.go]: Please specify the default if not given

File: main.go [hunk 4]
```diff
@@ -2857,6 +2893,45 @@ prints out information about a certificate, including:
 			}
 			fmt.Printf("\n")
 
+			fmt.Printf("  signed with the algorithm ")
+			var sigAlgo string
+			switch cert.Certificate.SignatureAlgorithm {
```
Comment (dennisjbell) [inline main.go]: Move this switch into x509 as GetSignatureLable (or similar) -- giant case statement doesn't belong in what is essentially a View

Comment (thomasmmitchell) [reply main.go]: Actually took this style from the key usage stuff right above it.

**Pull request merged**